src/network/networkd-link.c

```diff
@@ -1872,6 +1873,86 @@ static int link_up_can(Link *link) {
         return 0;
 }
 
+static int link_set_can(Link *link) {
+        _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *m = NULL;
+        int r;
+
+        assert(link);
+        assert(link->network);
+        assert(link->manager);
+        assert(link->manager->rtnl);
+
+        log_link_debug(link, "link_set_can");
+
+        r = sd_rtnl_message_new_link(link->manager->rtnl, &m, RTM_NEWLINK, link->ifindex);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Failed to allocate netlink message: %m");
+
+        r = sd_netlink_message_set_flags(m, NLM_F_REQUEST | NLM_F_ACK);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Could not set netlink flags: %m");
+
+        r = sd_netlink_message_open_container(m, IFLA_LINKINFO);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Failed to open netlink container: %m");
+
+        r = sd_netlink_message_open_container_union(m, IFLA_INFO_DATA, link->kind);
+        if (r < 0)
+                return log_link_error_errno(link, r, "Could not append IFLA_INFO_DATA attribute: %m");
+
+        if (link->network->can_bitrate || link->network->can_sample_point) {
+                struct can_bittiming bt = {
+                        .bitrate = link->network->can_bitrate,
+                        .sample_point = link->network->can_sample_point,
+                };
+
+                log_link_debug(link, "Setting bitrate = %d", bt.bitrate);
+                log_link_debug(link, "Setting sample point = %d", bt.sample_point);
+
+                r = sd_netlink_message_append_data(m, IFLA_CAN_BITTIMING, &bt, sizeof(bt));
+                if (r < 0)
+                        return log_link_error_errno(link, r, "Could not append IFLA_CAN_BITTIMING attribute: %m");
+        }
+
+        if (link->network->can_restart_us) {
```

Comment:
as above, please use explicit numeric comparisons for this...

Hmm, here's one question though: we try to be careful with time value of 0 these days. Some APIs are sloppy and say that a time value of zero equals "off", even though a time value of zero could conceptually also mean "right-away". Note that our usec parsers usually also accept parsing the string "infinity" into the special value USEC_INFINITY (which is (usec_t) -1)). Quite often allowing a zero value while saying "infinity" means "off" is the better approach. Is this a case like this?

Reply:
Then `restart_ms` is one these sloppy APIs. `restart_ms == 0` means never automatically restart, `USEC_INFINITY` would mean the same.